### ISSUE TITLE
Tokyo monorail route crash from 1.7.0.0

### ISSUE THREAD

**Issue by ginga81** [leezer3/OpenBVE#577]
### Description

From Japanese Slack comunity user, when we are attempt to load Tokyo Monorail route, OpenBVE is crash and back to the main menu.
At the 1.6.0.0, this route can run perfectly, but 1.7.0.0 or newer(Including 1.7.2.3), the route is crash.
Additionaly information, 1.4.3 can also drive.

## Route

BVE4/【普通】1327H.csv
https://web.archive.org/web/20110505123731/http://shtr-m.net/bve/tmk.html

## Train
Also can download Series-2000 at here.
https://web.archive.org/web/20110505123731/http://shtr-m.net/bve/tmk.html

## Logs
This log is version 1.7.0.0's.
openBVE Log: 2021/01/26 2:36:25

02:36:25  Initialising game window of size 960 x 600
02:36:26  Game window initialised successfully.
02:36:26  Loading route file: /home/user/ダウンロード/東京モノレールBVE24/railway/route/東京モノレール/BVE4/【普通】1327H.csv
02:36:26  INFO: Route file hash C292164821AA50539466F8C5F2C460B09F105E6085AC68D9D2B27CB65A6E4779
02:36:26  /home/user/ダウンロード/東京モノレールBVE24/railway : Railway folder found.
02:36:26  Route file format is: CSV
02:36:28  CycleIndex 0 references an object not loaded in Track.Ground at line 441, column 6 in file /home/user/ダウンロード/東京モノレールBVE24/railway/route/東京モノレール/BVE4/【普通】1327H.csv
02:36:28  The route and train loader encountered the following critical error: The given key '0' was not present in the dictionary.
02:36:28  WARNING: Program crashing. Creating CrashLog file: /home/user/.config/openBVE/Settings/OpenBVE Crash- 2021.1.26[02.36].log
***
OpenBVE Crash- 2021.1.26[02.36].log
2021/01/26 2:36:28
OpenBVE 1.7.0.0 Crash Log
Program is running on the Linux backend
Current screen resolution is: Windowed 960px X 600px 
Current routefile is: /home/user/ダウンロード/東京モノレールBVE24/railway/route/東京モノレール/BVE4/【普通】1327H.csv
Current train is: /home/user/ダウンロード/東京モノレールBVE24/Train/tmk2000
The current routefile caused the following exception: 
System.Collections.Generic.KeyNotFoundException: The given key '0' was not present in the dictionary.
  at System.Collections.Generic.Dictionary`2[TKey,TValue].get_Item (TKey key) [0x0001e] in <d13c8b563008422a8c5aaec0a74089cc>:0 
  at OpenBve.CsvRwRouteParser.ParseRouteForData (System.String FileName, System.Boolean IsRW, System.Text.Encoding Encoding, OpenBve.CsvRwRouteParser+Expression[] Expressions, System.Double[] UnitOfLength, OpenBve.CsvRwRouteParser+RouteData& Data, System.Boolean PreviewOnly) [0x16767] in <13dbd95abec84772af58aa71037bf9b5>:0 
  at OpenBve.CsvRwRouteParser.ParseRouteForData (System.String FileName, System.Boolean IsRW, System.Text.Encoding Encoding, OpenBve.CsvRwRouteParser+RouteData& Data, System.Boolean PreviewOnly) [0x0005e] in <13dbd95abec84772af58aa71037bf9b5>:0 
  at OpenBve.CsvRwRouteParser.ParseRoute (System.String FileName, System.Boolean IsRW, System.Text.Encoding Encoding, System.String trainPath, System.String objectPath, System.String soundPath, System.Boolean PreviewOnly) [0x00e1d] in <13dbd95abec84772af58aa71037bf9b5>:0 
  at OpenBve.Loading.LoadEverythingThreaded () [0x000a0] in <13dbd95abec84772af58aa71037bf9b5>:0 
  at OpenBve.Loading.LoadThreaded () [0x00000] in <13dbd95abec84772af58aa71037bf9b5>:0   at System.Environment.get_StackTrace () [0x00000] in <d13c8b563008422a8c5aaec0a74089cc>:0 
  at OpenBve.Loading.LoadThreaded () [0x00000] in <13dbd95abec84772af58aa71037bf9b5>:0 
  at System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) [0x00000] in <d13c8b563008422a8c5aaec0a74089cc>:0 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <d13c8b563008422a8c5aaec0a74089cc>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <d13c8b563008422a8c5aaec0a74089cc>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) [0x00000] in <d13c8b563008422a8c5aaec0a74089cc>:0 
  at System.Threading.ThreadHelper.ThreadStart () [0x00000] in <d13c8b563008422a8c5aaec0a74089cc>:0 
Current program memory usage: 501.34mb
System memory free: 0mb

### Related information 

My PC's SPEC:

OS: Ubuntu18.04LTS(64bit)
Memory:31.4 GiB
CPU:AMD® Ryzen 7 2700x eight-core processor × 16
Graphic board:Radeon RX 590 Series (POLARIS10, DRM 3.33.0, 5.3.0-51-generic, LLVM 9.0.0)
Not use Joysticks

**Comment by leezer3** [comment]
Thanks.

Will take a look later tonight, hopefully simple.

**Comment by leezer3** [comment]
Fixed, sorry about that.

A different fix got broken when converting the backgrounds to a list as opposed to an array.
TLDR:
This route doesn't define background 0
It's first .Back command is at 25m
This means that block 0 had an invalid background state.
Previously, I was just pulling the first element out of the array, but this doesn't work with lists, but throws the exception above.

Typically gets found hours after I put out a release.....
